### PR TITLE
ts-web/core: Run playground e2e in nodejs

### PR DESCRIPTION
**Pull request by lukaw3d**
In case we want to ensure ts-web/core works in nodejs too (with xhr2 polyfill)

### CODE REVIEW

**Comment by lukaw3d** [comment]
Ah, nevermind, it breaks at `crypto is not defined`

**Comment by lukaw3d** [comment]
node 18 says:
Error: callUnary method http://localhost:42280/oasis-core.NodeController/WaitReady: RpcError: Http response at 400 or 500 level, http status code: 0

**Comment by pro-wh** [comment]
is that on public node? public node doesn't allow Wait* calls

**Comment by lukaw3d** [comment]
nope http://localhost:42280/oasis-core.NodeController/WaitReady

**Comment by pro-wh** [comment]
does xhr have a debug mode? grpc-web is bad about always saying "400 or 500 level" even when it's not that :slightly_frowning_face:

**Comment by pro-wh** [comment]
does the WaitReady work in local testing?

**Comment by pro-wh** [comment]
oh it's complaining about stakingDelegationsTo now, weird

**Comment by pro-wh** [comment]
> HTTP 1937: SOCKET ERROR: connect ECONNREFUSED 127.0.0.1:42280 Error: connect ECONNREFUSED 127.0.0.1:42280

why :skull:

**Comment by pro-wh** [comment]
> [2023-05-09 21:01:49.358][1833][critical][main] [source/server/server.cc:131] error initializing configuration 'sample-envoy.yaml': Protobuf message (type envoy.config.bootstrap.v3.Bootstrap reason INVALID_ARGUMENT:(route_config.virtual_hosts[0].routes[1]) prefix: Cannot find field.) has unknown fields
[2023-05-09 21:01:49.358][1833][info][main] [source/server/server.cc:980] exiting
Protobuf message (type envoy.config.bootstrap.v3.Bootstrap reason INVALID_ARGUMENT:(route_config.virtual_hosts[0].routes[1]) prefix: Cannot find field.) has unknown fields

oh shoot there's stuff in the start proxy step

**Comment by lukaw3d** [comment]
idk
start proxy shouldn't matter, because cypress was passing. I can reenable cypress to show the api works

File: client-sdk/ts-web/core/playground/sample-envoy.yaml [hunk 1]
```diff
@@ -18,16 +18,15 @@ static_resources:
...
+                safe_regex:
+                  regex: '/oasis-core\.(Registry/(GetNodes)|Staking/(Account|DebondingDelegationInfosFor|DelegationInfosFor|DelegationsTo|GetEvents)|Consensus/(SubmitTx|EstimateGas|GetSignerNonce|GetBlock|GetTransactionsWithResults|GetGenesisDocument|GetChainContext|WatchBlocks)|NodeController/(WaitReady))'
                 route:
                   cluster: oasis_node_grpc
                   timeout: 0s
                   max_stream_duration:
                     grpc_timeout_header_max: 0s
               - match:
-                  prefix: '/oasis-core'
+                prefix: '/oasis-core'
```
Comment (pro-wh) [inline client-sdk/ts-web/core/playground/sample-envoy.yaml:29]: :flashlight: indentation breakage

**Comment by lukaw3d** [comment]
unless cypress tests have been a lie all along

**Comment by pro-wh** [comment]
> start proxy shouldn't matter, because cypress was passing.

from this one? no "start proxy" complaints in this one
https://github.com/oasisprotocol/oasis-sdk/actions/runs/4812687701/jobs/8568248031

**Comment by lukaw3d** [comment]
oh. Cypress is failing too! https://github.com/oasisprotocol/oasis-sdk/actions/runs/4941702070/jobs/8834600915

**Comment by lukaw3d** [comment]
okay, summary time:
- before bded9eb030e7eba0b06e9e77e1936f1c4b69d822: cypress passed, nodejs 14 passed until crypto was needed
- bded9eb030e7eba0b06e9e77e1936f1c4b69d822: cypress passed, nodejs 18 immediately failed
- ba55f87bf425f3a869aea675983921a525798326: i removed cypress in case it's requests broke the API. They didn't
- 662ab03154dc09476ebdd68493343f36634c3191: nodejs 18 worked against public grpc :shrug: 
- e2918ee11018a8bc345114e485c250967ebb824c: i attempted to upgrade envoy. I probably broke it. And I didn't have cypress to tell me
- 1b9862d09c12565835c56f1e9ffb6be6f2c959e2: enabled nagle's algorithm (defaulted to disabled in nodejs16). Doesn't seem to matter
- f94f88a76bb7daabc690cb9c7766280d53b12895: --dns-result-order=ipv4first (defaulted to verbatim in nodejs17). This would have worked if I didn't break envoy earlier.

175a05dd4c00708c399423373a7ed472ac7a486d --dns-result-order=ipv4first works because I guess envoy doesn't like ipv6. And all of it would have worked against public grpc since the beginning

**Comment by lukaw3d** [comment]
update: https://github.com/oasisprotocol/oasis-sdk/pull/1246/commits/9d74cd5645e90afd731495cc5508e2e083fc39c0 nodejs 18 only passed until crypto was needed :facepalm:

**Comment by pro-wh** [comment]
oh shoot, the envoy config specifically says the ipv4 address 127.0.0.1. maybe we should change `localhost` in the test to 127.0.0.1

**Pull request closed without merging**